client/my-sites/checkout/checkout-thank-you/index.jsx

```diff
@@ -486,17 +492,22 @@ export class CheckoutThankYou extends Component {
 					<PlanThankYouCard siteId={ this.props.selectedSite.ID } { ...planProps } />
 				</Main>
 			);
-		} else if ( wasDomainMappingOrTransferProduct ) {
-			const [ purchaseType, predicate ] = purchases.some( isDomainMapping )
-				? [ 'MAPPING', isDomainMapping ]
-				: [ 'TRANSFER', isDomainTransfer ];
+		} else if ( wasDomainMappingOrTransferProduct || wasDomainMappingOrRegistration ) {
+			const [ purchaseType, predicate ] = this.domainPurchaseType( purchases );
 			const [ , domainName ] = findPurchaseAndDomain( purchases, predicate );
+
+			const inboxPurchase = this.getProfessionalEmailFromPurchase(
+				wasDomainMappingOrRegistration,
+				purchases
+			);
```

Comment:
If we shift to a single `wasDomainProduct` variable, we can clean this logic up as follows:
```suggestion
		} else if ( wasDomainProduct ) {
			const [ purchaseType, predicate ] = this.domainPurchaseType( purchases );
			const [ , domainName ] = findPurchaseAndDomain( purchases, predicate );

			const inboxPurchase = this.getProfessionalEmailFromPurchase(
				purchaseType,
				purchases
			);
```